app/src/main/kotlin/at/bitfire/davdroid/sync/JtxSyncManager.kt

```diff
@@ -163,9 +165,14 @@ class JtxSyncManager @AssistedInject constructor(
 
         icalobjects.forEach { jtxICalObject ->
             // if the entry is a recurring entry (and therefore has a recurid)
-            // we udpate the existing (generated) entry
+            // we update the existing (generated) entry
             if(jtxICalObject.recurid != null) {
-                val local = localCollection.findRecurring(jtxICalObject.uid, jtxICalObject.recurid!!, jtxICalObject.dtstart!!)
+                val local = jtxICalObject.dtstart?.let { dtstart ->
+                    localCollection.findRecurring(jtxICalObject.uid, jtxICalObject.recurid!!, dtstart)
+                } ?: run {
+                    logger.log(Level.INFO, "Got a recur instance (${jtxICalObject.recurid}/${jtxICalObject.uid}) with an invalid dtstart.")
+                    null
```

Comment:
`null` here will make `local = null`. So it will be added by `newLocal.add()` below.

However I wonder whether this is what we want in all cases. What happens when the task already exists? Like this:

1. Add task with UID `12345` on server.
2. Sync.
3. Change task on the server so that it has an `RRULE` but no `DTSTART`.
4. Sync.
5. The new code will set `val local = null` and thus "add" the task recurrence locally although it's already there. It should be updated or completely ignored instead.

This is what I meant in the second list item of https://github.com/bitfireAT/davx5-ose/pull/1336#issuecomment-2706478263.

Maybe I have overlooked something …

Reply:
Why don't we always query for the recurrence instance without DTSTART? By the standard it would suffice to use UID and RECURRENCE-ID, no? 

It would make this so much easier ... 
We would have to make a change in ical4android then though.

Reply:
Here's the PR: https://github.com/bitfireAT/ical4android/pull/195

Reply:
Merged